File: backend/src/main/java/cliquee/backend/model/Event.java
package cliquee.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;
import lombok.Data;
import org.hibernate.annotations.UuidGenerator;

@Data
@Entity
@Table(name = "events")
public class Event {

  @Id
  @UuidGenerator
  @Column(updatable = false, nullable = false)
  private UUID id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "group_id", nullable = false)
  private Group group;

  private String name;
  private String description;
  private String location;
  private LocalDateTime start_time;
  private LocalDateTime end_time;
  private LocalDateTime created_at = LocalDateTime.now();
  private LocalDateTime updated_at = LocalDateTime.now();

  @OneToMany(mappedBy = "event")
  private List<EventParticipant> eventParticipants;

  @OneToMany(mappedBy = "event")
  private List<EventComment> eventComments;
}

File: flaskBackend/app/models/event.py
from datetime import datetime
from ..extensions import db

class Event(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    group_id = db.Column(db.Integer, db.ForeignKey('group.id'), nullable=False)
    name = db.Column(db.String(64), index=True, unique=False)
    description = db.Column(db.String(64))
    location = db.Column(db.String(64))
    start_time = db.Column(db.DateTime)
    end_time = db.Column(db.DateTime)
    created_at = db.Column(db.DateTime, default=datetime.now())
    updated_at = db.Column(db.DateTime, default=datetime.now(), onupdate=datetime.now())

    event_participants = db.relationship('Event_Participant', backref='event', lazy=True)
    event_comments = db.relationship('Event_Comment', backref='event', lazy=True)

    def __repr__(self):
        return f'<Event {self.name}>'